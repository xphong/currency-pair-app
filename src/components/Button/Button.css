.button {
  background: blue;
  color: #fff;
  border: none;
  padding: 24px;
  font-size: 16px;
  width: 100%;
  cursor: pointer;
}
