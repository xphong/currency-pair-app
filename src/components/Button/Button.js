import React from 'react';
import PropTypes from 'prop-types';

import styles from './Button.css';

const Button = props => {
  const { handleOnClick, children } = props;

  return (
    <button className={styles.button} onClick={handleOnClick}>
      {children}
    </button>
  );
};

Button.propTypes = {
  handleOnClick: PropTypes.func
};

export default Button;
