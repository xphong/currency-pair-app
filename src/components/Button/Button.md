### Default Style

```js
<Button>Submit</Button>
```
