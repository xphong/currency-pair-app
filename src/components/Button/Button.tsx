import React from 'react';

import styles from './Button.css';

type ButtonProps = {
  handleOnClick: () => void;
  children: React.ReactNode;
}

const Button = (props: ButtonProps) => {
  const { handleOnClick, children } = props;

  return (
    <button className={styles.button} onClick={handleOnClick}>
      {children}
    </button>
  );
};

export default Button;
