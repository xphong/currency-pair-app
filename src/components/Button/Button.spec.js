import React from 'react';

import Button from './Button';

describe('Button Component', () => {
  it('should render Button', () => {
    const wrapper = shallow(<Button />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should render Button with handleOnClick', () => {
    const handleClick = jest.fn();
    const wrapper = shallow(<Button handleClick={handleClick}>It's over 9000!!</Button>);

    expect(wrapper).toMatchSnapshot();
  });
});
