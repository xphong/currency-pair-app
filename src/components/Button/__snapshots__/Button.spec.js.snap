// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component should render Button 1`] = `
<button
  className="button"
/>
`;

exports[`Button Component should render Button with handleOnClick 1`] = `
<button
  className="button"
>
  It's over 9000!!
</button>
`;
